### PR TITLE
changed instances of ffc-pay to ffc-doc

### PR DESCRIPTION
**Pull request by GodsonLeigh**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-ARG PARENT_VERSION=2.2.1-node20.9.0
-ARG PORT=3000
+ARG PARENT_VERSION=2.1.2-node18.11.0
```
Comment (samplackett) [inline Dockerfile:1]: Any particular reason we want to downgrade the node version?

File: app/config/storage.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const Joi = require('joi')
+
+// Define config schema
```
Comment (samplackett) [inline app/config/storage.js]: Remove comment

File: app/config/storage.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const schema = Joi.object({
+  connectionStr: Joi.string().when('useConnectionStr', { is: true, then: Joi.required(), otherwise: Joi.allow('').optional() }),
+  storageAccount: Joi.string().required(),
+  container: Joi.string().default('statements'),
+  folder: Joi.string().default('outbound'),
+  useConnectionStr: Joi.boolean().default(false),
+  createContainers: Joi.boolean().default(true)
+})
+
+// Build config
```
Comment (samplackett) [inline app/config/storage.js]: Remove comment

File: app/config/storage.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const config = {
+  connectionStr: process.env.AZURE_STORAGE_CONNECTION_STRING,
+  storageAccount: process.env.AZURE_STORAGE_ACCOUNT_NAME,
+  container: process.env.AZURE_STORAGE_CONTAINER,
+  folder: process.env.AZURE_STORAGE_FOLDER,
+  useConnectionStr: process.env.AZURE_STORAGE_USE_CONNECTION_STRING,
+  createContainers: process.env.AZURE_STORAGE_CREATE_CONTAINERS
+}
+
+// Validate config
```
Comment (samplackett) [inline app/config/storage.js]: Remove comment

File: app/config/storage.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+const result = schema.validate(config, {
+  abortEarly: false
+})
+
+// Throw if config is invalid
```
Comment (samplackett) [inline app/config/storage.js]: Remove comment

File: app/plugins/errors.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+  plugin: {
+    name: 'errors',
+    register: (server) => {
+      server.ext('onPreResponse', (request, h) => {
+        const response = request.response
+
+        if (response.isBoom) {
+          // An error was raised during
+          // processing the request
```
Comment (samplackett) [inline app/plugins/errors.js]: Remove comment

File: app/plugins/errors.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+          const statusCode = response.output.statusCode
+
+          // Log the error
```
Comment (samplackett) [inline app/plugins/errors.js]: Remove comment

File: docker-compose.link.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+networks:
+  default:
+    external:
+      name: ffc-pay
```
Comment (samplackett) [inline docker-compose.link.yaml:4]: does this need to be ffc-doc?

File: docs/openapi.yaml [hunk 9]
```diff
@@ -0,0 +1,40 @@
+openapi: '3.0.0'
+
+info:
+  title: FFC Pay Statement Receiver
```
Comment (samplackett) [inline docs/openapi.yaml]: FFC Doc

File: scripts/start [hunk 10]
```diff
@@ -1,4 +1,14 @@
 #!/usr/bin/env sh
+if [ -z "$(docker network ls --filter name=^ffc-pay$ --format={{.Name}})" ]; then
+  echo "Creating ffc-pay Docker network"
+  docker network create ffc-pay
```
Comment (samplackett) [inline scripts/start:4]: do these docker scripts need to create the ffc-doc network instead of pay?

File: Dockerfile (hunk 1 above)
Comment (GodsonLeigh) [reply Dockerfile:1]: For now, only to keep it in line with what exists in ffc-pay. As John set this up using the template I have not upgraded the node version on our existing version in order to check that it all still works.

File: docker-compose.link.yaml (hunk 8 above)
Comment (GodsonLeigh) [reply docker-compose.link.yaml:4]: All of the other services have kept the ffc-pay network so I left this as-is

File: scripts/start (hunk 10 above)
Comment (GodsonLeigh) [reply scripts/start:4]: This is the network that everything runs on together so it was left as ffc-pay

File: package.json [hunk 11]
```diff
@@ -23,7 +23,7 @@
   ],
   "license": "OGL-UK-3.0",
   "dependencies": {
-    "@azure/identity": "2.0.5",
+    "@azure/identity": "^4.4.0",
```
Comment (samplackett) [inline package.json]: our package versions should be hard coded rather than ^

File: package.json [hunk 12]
```diff
@@ -36,7 +36,7 @@
     "blipp": "4.0.2",
     "jest": "28.1.2",
     "jest-junit": "14.0.0",
-    "nodemon": "2.0.19",
+    "nodemon": "^3.1.4",
```
Comment (samplackett) [inline package.json]: as above

**Review by samplackett — APPROVED** [review]
200w

**Pull request merged**